### DISCUSSION TITLE
Missing X-Grafana-Org-Id and other headers?

### DISCUSSION THREAD

**Discussion by kfdm** [grafana/grafana#40666] [Data Sources]
When providing Grafana and Prometheus as a service to other teams, it's often necessary to track down where a query is coming from for debugging and diagnostic purposes. When I query from a dashboard directly, I get these headers

```
X-Dashboard-Id
X-Grafana-Org-Id
X-Panel-Id
```

which can often be helpful for debugging. Thanos (and I presume Cortext) also can use these headers for help in query logging (example `--query-frontend.org-id-header=<http-header-name>`) 

However, when alerting is configured on a dashboard, even though they query the same URls from the data source's perspective (example `/api/v1/query_range`) none of these extra headers are sent, which then requires one to force the headers to be included from the Datasource settings directly, as a custom header.

This feels like a bug (inconsistencies regarding how the same endpoint is called) but I'm not sure if this is something that will be covered by #37784 or is another bug. 

I've tried following some of the code, but eventually get fairly confused as it's not very easy to track where these `X-` headers are set. I would appreciate any pointers on where I should be looking, and if this should be reported as a bug or not.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
